### PR TITLE
Handle repo connect event on - repo connect command

### PR DESCRIPTION
**Pull request by neha-Gupta1**
## Description

Create a `repo connect` event on CLI's repo connect command 

## Does this PR need a docs update or release note?

- [ ] :white_check_mark: Yes, it's included
- [ ] :clock1: Yes, but in a later PR
- [ ] :no_entry: No 

## Type of change

- [x] :sunflower: Feature

## Issue(s)
https://github.com/alcionai/corso/issues/1762

## Test Plan

- [x] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: src/pkg/control/options.go [hunk 1]
```diff
@@ -11,6 +11,7 @@ type Options struct {
 	FailFast           bool            `json:"failFast"`
 	RestorePermissions bool            `json:"restorePermissions"`
 	ToggleFeatures     Toggles         `json:"ToggleFeatures"`
+	RepoConnect        bool            `json:"-"`
```
Comment (ryanfkeepers) [inline src/pkg/control/options.go:14]: I think `options` is the wrong owner for this flag.  The struct is used to aggregate runtime configurations: settings that could cascade to many different places throughout processing.  What you're solving in this PR is a one-off flag that only involves the repository.Connect call.  Placing the flag into options will expose it to a lot more topology, and that might confuse contributors who are working with options downstream.

My recommendation is to add the flag to the Connect() parameters themselves.  That way it's clear that this change only affects that one function call, and isn't going to cause confusion elsewhere.

File: src/cli/repo/s3.go [hunk 2]
```diff
@@ -183,7 +183,7 @@ func connectS3Cmd(cmd *cobra.Command, args []string) error {
 		return Only(ctx, errors.Wrap(err, "Failed to parse m365 account config"))
 	}
 
-	r, err := repository.Connect(ctx, a, s, options.Control())
+	r, err := repository.Connect(ctx, a, s, true, options.Control())
```
Comment (vkamra) [inline src/cli/repo/s3.go:186]: @ryanfkeepers @neha-Gupta1 - thoughts on just adding a `func ConnectWrapper(...)` helper in the `repo` package that connects to the repo and fires the event? Seems cleaner.

Comment (neha-Gupta1) [reply src/cli/repo/s3.go:186]: Seems to be a cleaner way. If @ryanfkeepers agrees with it.

Comment (vkamra) [reply src/cli/repo/s3.go:186]: @neha-Gupta1 - to save you the wait till tomorrow - how about we move forward with that approach. I think @ryanfkeepers will be on board and if not I will owe you both some ☕

**Review by ryanfkeepers — APPROVED** [review]

File: src/cli/repo/s3.go (hunk 2 above)
Comment (ryanfkeepers) [reply src/cli/repo/s3.go]: I am in such deep coffee debt.

**Pull request merged**